src/Content/Version.php

```diff
@@ -427,9 +427,13 @@ protected function previewTokenFromUrl(string $url): string|null
 	{
 		$localPrefix = $this->model->kirby()->url('base') . '/';
 
-		// Todo: this is only a quick fix to get home page previews working again,
-		// we need to double-check if this is still correct
-		if (Str::startsWith($url, $localPrefix) === false && $url . '/' !== $localPrefix) {
+		// normalize homepage URLs to have a trailing slash
+		// to make the following logic work with those as well
+		if ($url . '/' === $localPrefix) {
+			$url .= '/';
+		}
+
+		if (Str::startsWith($url, $localPrefix) === false) {
 			return null;
 		}
```

Comment:
What do you think about following usage? I think it's more simply and readable.

````php
$baseUrl     = $this->model->kirby()->url('base');
$localPrefix = $baseUrl . '/';

if (Str::startsWith($url, $baseUrl) === false) {
    return null;
}
````

Reply:
I had that first, but it would break for subfolder installations. E.g. the current site is at `https://example.com/subfolder` and you have a preview link set to `https://example.com/subfolder-with-some-extra`. This URL starts with the index URL but points to a different folder and is not part of the same Kirby site.

Reply:
I guess I didn't explain it. You check by adding `/` to the `base` url and `$url`. I say let's check without adding `/` to both. 

What I mean is why don't we check without adding `/` at the end? Does the `/` at the end have anything to do with the different subfolder setup?

Reply:
I think so, because the `Str::startsWith()` check will return `true` for both even though one is considered external.